rfc-116-store-attachment-data-in-content-items.md

```diff
@@ -0,0 +1,391 @@
+     - `preview_url`
+   - `image_asset`
+     - `alt_text`
+     - `caption`
+     - `credit`
+
+   **Migration concerns:** This can be done without breaking
+   compatibility, as all new fields are optional.
+
+1. Add new attachment types for HTML and external attachments:
+
+   - `html_attachment_asset`:
+     - `attachment_type` (required; enum with permitted values `"html"`)
+     - `id`
+     - `locale`
+     - `title`
+     - `url` (required)
+   - `external_attachment_asset`:
+     - `attachment_type` (required; enum with permitted values `"external"`)
+     - `id`
+     - `locale`
+     - `title`
+     - `url` (required)
+
+   **Migration concerns:** This can be done without breaking
+   compatibility, as the new types are unused for now.
+
+1. Add new attachment type for publication attachments:
+
+    `publication_attachment_asset` (one-of combination of `file_attachment_asset`, `html_attachment_asset`, and `external_attachment_asset`):
+    - `command_paper_number`
+    - `hoc_paper_number`
+    - `isbn`
+    - `parliamentary_session`
+    - `unique_reference`
+    - `unnumbered_command_paper`
+    - `unnumbered_hoc_paper`
+
+   I've left out the following fields because they can be deduced from
+   others: `csv?`, `external?`, `external_url`, `file_extension`, `html?`,
+   `opendocument?`, `pdf?`
+
+   And these fields because they don't seem to be used:
+   `print_meta_data_contact_address`, `web_isbn`.
+
+   And these fields because the Publishing Workflow team are planning
+   to remove them from Whitehall: `order_url`, `price_in_pence`.
+
+   And these fields because they don't need to be exposed outside the
+   publishing app: `attachable_id`, `attachment_data_id`, `deleted`,
+   `ordering`, `slug`.
+
+   **Migration concerns:** This can be done without breaking
+   compatibility, as the new type will be unused for now.
+
+1. Make `attachment_type` and `id` in `file_attachment_asset` mandatory.
+
+    **Migration concerns:** this would need to be done in three steps:
+    1. Update specialist publisher, travel advice publisher, and
+       manuals publisher to set the fields.
+    2. Republish all affected documents.
+    3. Make the field mandatory.
+
+1. Add a *mandatory* `details.attachments` field to all formats which
+   can have attachments.
+
+    **Migration concerns:** this would need to be done in four steps:
+    1. Add the field as optional.
+    2. Update publishing apps to write to it.
+    3. Republish all affected documents.
+    4. Make the field mandatory.
+
+1. Add a *mandatory* `details.featured_attachments` list to
+   publication formats, which is an ordered list of govspeak or HTML
+   strings (using the `#/definitions/multiple_content_types` schema)
+   rendering attachments from `details.attachments`.
+
+   **Migration concerns:** this would need to be done in four steps:
+   1. Add the field as optional.
+   2. Update publishing apps (just Whitehall and Content Publisher) to write to it.
```

Comment:
Just a minor thing to check: we're intentionally leaving this open as to whether the app, or the Publishing API, does the actual rendering?

Reply:
I think it would be nice if the Publishing API rendered it, for consistency between apps, but ultimately it doesn't matter too much so long as all the metadata is still in `details.attachments`.